src/cata_tiles.cpp

```diff
@@ -533,34 +555,17 @@ void cata_tiles::draw(int destx, int desty, int centerx, int centery, int width,
     screentile_width = (width + tile_width - 1) / tile_width;
     screentile_height = (height + tile_height - 1) / tile_height;
 
-    for (int my = 0; my < sy; ++my) {
-        for (int mx = 0; mx < sx; ++mx) {
-            x = mx + o_x;
-            y = my + o_y;
-            l = light_at(x, y);
-            const auto critter = g->critter_at( x, y );
-            if (l != CLEAR) {
-                // Draw lighting
-                draw_lighting(x, y, l);
-                if( critter != nullptr && g->u.sees_with_infrared( *critter ) ) {
-                    draw_from_id_string( "infrared_creature", C_NONE, empty_string, x, y, 0, 0 );
-                }
-                continue;
-            }
-            // light is no longer being considered, for now.
-            // Draw Terrain if possible. If not possible then we need to continue on to the next part of loop
-            if (!draw_terrain(x, y)) {
-                continue;
-            }
-            draw_furniture(x, y);
-            draw_trap(x, y);
-            draw_field_or_item(x, y);
-            draw_vpart(x, y);
-            if( critter != nullptr ) {
-                draw_entity( *critter, x, y );
-            }
-        }
-    }
+    auto map_draw_function = [&](int x, int y, lit_level ll) -> void {
+        draw_specific_tile(x, y, ll);
+    };
+
+    g->m.draw_loop(
```

Comment:
well, this isn't JUST in service to the other PR. it will make additional renderers easier to implement. when someone adds cata_isometric_tiles, they don't have to re-implement this stuff in a third place.

They should be merged because they are doing the same job. "loop over the visible subset of the reality bubble, figure out which tiles I can see at which light levels, and draw them appropriately". The only difference is what "draw them appropriately" means, which is what the draw_func callback is for.

Ignoring the benefits of deduplicating the loop... caching the apparent_light values for the reality bubble would involve a 75kB array (less if we pack the lit_level down to fewer bits, but that's a lot of coding overhead and possible code slowness).